fusedrug/data/protein/structure/protein_complex.py

```diff
@@ -22,18 +22,26 @@ def __init__(self, verbose: bool = True) -> None:
         self.chains_data = {}  # maps from chain description (e.g. ('7vux', 'A')) to
         self.flattened_data = {}
 
+        #
+        self.per_chain_most_frequent_residue_part = {}
+        self.per_chain_mmcif_object = {}
+        self.per_chain_mmcif_dict = {}
+
     def add(
         self,
-        pdb_id: str,
+        pdb_id_or_filename: str,
+        pdb_id: Optional[str] = None,
```

Comment:
Why do you keep pdb_id? Is backward compatibility important here?

Reply:
pdb_id_or_filename might be a path to a local pdb file, so pdb_id gives the user an option to define the pdb_id as there is no implemented way to automatically extract it.

I'll check ways to extract it automatically from the pdb file itself, and check if it's important to allow the user to override that.

That will be a separate PR.

about what you asked - backward compatibility isn't super critical here, as we can likely detect all usage locations and port them to newer usage.